### PR TITLE
Move to AVA for testing

### PR DESCRIPTION
**Pull request by tdeschryver**
With this change all tests are being run with `AVA` instead of `mocha`.

### CODE REVIEW

File: test/ask-permission.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import {spawn} from 'child_process';
+import test from 'ava';
+
+test.cb('skip in TTY mode', t => {
+	t.plan(1);
+
+	const insProcess = spawn('node', [
```
Comment (kevva) [inline test/ask-permission.js]: I think you can use [`execa`](https://github.com/sindresorhus/execa) instead.

```
test('skip in TTY mode', async t => {
	const {code} = await execa('node', [./test/fixtures/sub-process.js]);
	t.is(code, 145);
});
```

Do it on the the following tests too.

File: test/ask-permission.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		t.is(code, 145);
+		t.end();
+	});
+});
+
+test.cb('skip in CI mode', t => {
+	t.plan(1);
+
+	const env = JSON.parse(JSON.stringify(process.env));
+	env.CI = true;
```
Comment (kevva) [inline test/ask-permission.js]:
```
const CI = process.env.CI;
process.env.CI = true;

... the test ...

process.env.CI = CI;
```

File: test/ask-permission.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		t.is(code, 145);
+		t.end();
+	});
+});
+
+test.cb('skip after timeout', t => {
+	t.plan(1);
+
+	const env = JSON.parse(JSON.stringify(process.env));
+	env.permissionTimeout = 0.1;
```
Comment (kevva) [inline test/ask-permission.js]: Same as the `CI` test.

File: test/insight.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import objectValues from 'object-values';
```
Comment (kevva) [inline test/insight.js]: You can use `Object.values` instead.

File: test/insight.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	packageName: 'yeoman',
+	packageVersion: '0.0.0'
+});
+
+insight.optOut = false;
+
+test('put tracked path in queue', t => {
+	Insight.prototype._save = function () {
+		t.is('/test', objectValues(this._queue)[0].path);
+	};
```
Comment (kevva) [inline test/insight.js]:
```js
Insight.prototype._save = () => {
    t.is('/test', Object.values(this._queue)[0].path);
};
```

File: test/providers.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+
```
Comment (kevva) [inline test/providers.js]: Remove this.

File: test/providers.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+const ver = '0.0.0';
+
+let config;
+let insight;
+
+test.beforeEach(() => {
+	config = {
+		get: sinon.spy(() => {
+			return true;
+		}),
```
Comment (kevva) [inline test/providers.js]: Since you `return` directly, do this instead:

```js
get: sinon.spy(() => true),
```

File: test/providers.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+	insight = new Insight({
+		trackingCode: 'xxx',
+		packageName: pkg,
+		packageVersion: ver,
+		config: config
+	});
+});
+
+test('access the config object for reading', t => {
+	t.truthy(insight.optOut);
```
Comment (kevva) [inline test/providers.js]: What's the returned value from `insight.optOut`? Prefer to use `t.true` or `t.is`.

File: test/providers.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+	t.truthy(config.get.called);
+});
+
+test('access the config object for writing', t => {
+	const sentinel = {};
+	insight.optOut = sentinel;
+	t.truthy(config.set.calledWith('optOut', sentinel));
```
Comment (kevva) [inline test/providers.js]: Same as the above.

File: test/providers.js (hunk 8 above)
Comment (tdeschryver) [inline test/providers.js]: Its returning true, I'll change it to `t.true`. Thanks for the review!

**Comment by kevva** [comment]
Good job :+1:

File: test/ask-permission.js (hunk 2 above)
Comment (kevva) [inline test/ask-permission.js]: With `execa` you don't need to pass in the `env`. So either do `execa('node', args, {extendEnv: false, env: {CI: true})` or set `process.env` like my example before the call.

File: test/ask-permission.js (hunk 1 above)
Comment (tdeschryver) [inline test/ask-permission.js]: Just to be sure, if I do this the `execa` will be rejected. It is OK to wrap it with a `try catch` right?
```
test('skip in TTY mode', async t => {
	try {
		await execa('node', ['./test/fixtures/sub-process.js']);
	} catch (err) {
		t.is(err.code, 145);
	}
});
```

Comment (kevva) [inline test/ask-permission.js]: Oh, do `const err = await t.throws(execa(...));`.

Comment (tdeschryver) [inline test/ask-permission.js]: Ah, that's better, thanks!

File: test/insight.js (hunk 4 above)
Comment (tdeschryver) [inline test/insight.js]: Seems like this crashes on `node.js` 4 and 6. OK if I revert to `object-values`?

Comment (kevva) [inline test/insight.js]: Hmm, alright. Thought that was transpiled.

**Comment by sindresorhus** [comment]
Looks great! :)

**Pull request merged**